cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ConnectionUtils.java

```diff
@@ -0,0 +1,139 @@
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.macro.MacroEvaluator;
+import io.cdap.cdap.api.macro.MacroParserOptions;
+import io.cdap.cdap.api.plugin.InvalidPluginConfigException;
+import io.cdap.cdap.api.plugin.PluginProperties;
+import io.cdap.cdap.api.service.http.ServicePluginConfigurer;
+import io.cdap.cdap.datapipeline.connection.DefaultConnectorContext;
+import io.cdap.cdap.datapipeline.connection.LimitingConnector;
+import io.cdap.cdap.etl.api.batch.BatchConnector;
+import io.cdap.cdap.etl.api.connector.Connector;
+import io.cdap.cdap.etl.api.connector.ConnectorContext;
+import io.cdap.cdap.etl.api.connector.ConnectorSpec;
+import io.cdap.cdap.etl.api.connector.DirectConnector;
+import io.cdap.cdap.etl.api.connector.SampleRequest;
+import io.cdap.cdap.etl.proto.ArtifactSelectorConfig;
+import io.cdap.cdap.etl.proto.connection.ConnectionBadRequestException;
+import io.cdap.cdap.etl.proto.connection.ConnectorDetail;
+import io.cdap.cdap.etl.proto.connection.PluginDetail;
+import io.cdap.cdap.etl.proto.connection.PluginInfo;
+import io.cdap.cdap.etl.proto.connection.SampleResponse;
+import io.cdap.cdap.etl.proto.validation.SimpleFailureCollector;
+import io.cdap.cdap.etl.spec.TrackedPluginSelector;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import javax.ws.rs.BadRequestException;
+
+public class ConnectionUtils {
+
+  private ConnectionUtils() {
+    // prevent instantiation of util class.
+  }
+
+  /**
+   * Returns {@link Connector}
+   * @param configurer {@link ServicePluginConfigurer}
+   * @param pluginInfo {@link PluginInfo}
+   * @param pluginSelector {@link TrackedPluginSelector}
+   * @param macroEvaluator {@link MacroEvaluator}
+   * @param options {@link MacroParserOptions}
+   * @return {@link Connector}
+   */
+  public static Connector getConnector(ServicePluginConfigurer configurer, PluginInfo pluginInfo,
+                                       TrackedPluginSelector pluginSelector, MacroEvaluator macroEvaluator,
+                                       MacroParserOptions options) {
+    Connector connector = null;
+    try {
+      connector = configurer.usePlugin(pluginInfo.getType(), pluginInfo.getName(), UUID.randomUUID().toString(),
+                                       PluginProperties.builder().addAll(pluginInfo.getProperties()).build(),
+                                       pluginSelector, macroEvaluator, options);
+    } catch (InvalidPluginConfigException e) {
+      throw new ConnectionBadRequestException(
+        String.format("Unable to instantiate connector plugin: %s", e.getMessage()), e);
+    }
+
+    if (connector == null) {
+      throw new ConnectionBadRequestException(String.format("Unable to find connector '%s'", pluginInfo.getName()));
+    }
+    return connector;
+  }
+
+  /**
+   * Return {@link SampleResponse} for the connector
+   *
+   * @param connector {@link Connector}
```

Comment:
Can just remove all these params since their name is understandable

Reply:
done